### PR TITLE
Change SchemaGroup to act as dict for determining search service, update documentation

### PR DESCRIPTION
**Pull request by ivana-truong**
Continues on work in PR #44, PR #45

SchemaGroup
- Removed rcsb_attributes_dict but kept new methods/overloaded other methods to allow SchemaGroup to contain the same information and also be a dictionary
- Added method to return list of all attributes
- Making an AttributeQuery with "rcsb_id" and no search service will throw an error. "attrs.rcsb_id" will also throw an error and say make an AttributeQuery and specify service.

Documentation
- Rewrote README and moved previous README info into readthedocs "Query" page
- Reorganized readthedocs and added a few pages
    - Attribute searching methods
    - Moved Sequence Motif Search, Facets, etc into "Additional Examples" page

### CODE REVIEW

**Review by piehld — COMMENTED** [review]
Thanks @ivana-truong! Your changes overall look excellent. Also great job on the documentation and cleaning up the usage of the CONSTS in public examples. It definitely looks a lot better now.

I just had a few comments/questions here, but in a minute I'm going to be creating a PR from my branch into yours with some updates to the documentation. I reorganized it a bit and renamed some sections. Feel free to review that PR and merge it when possible, but there are some additional changes I'd like to request after doing so...

1. Can you add example usage of the `rcsb_query_editor_url()` and `rcsb_query_builder_url()` methods under "Sessions"?
2. For the new "Search Service Types" section, the general focus of that should be on what each service type is used for and what its input arguments (and possible & default values) are, and less so on the scientific context of the example. Currently, I've just copied the first example of each service type (from `additional_examples.md`) into its corresponding section in "Search Service Types" just to get things organized as desired, so it's still kind of incomplete. Would you be able to provide the list of arguments for each type in that section?

Last, on a separate note, I did run into one error when playing around with the package, specifically when trying to create a subquery using the operator syntax for "exists", i.e.:

```python

from rcsbsearchapi import rcsb_attributes as attrs

q = bool(attrs.drugbank_info.drug_groups)  

# or more naturally

q = not attrs.drugbank_info.drug_groups

```
I'm pretty sure this should be returning an `AttributeQuery` object with the `exists` operator, but Python isn't liking that. Wonder if you can try to take a look?

Thanks again!

**Comment by piehld** [comment]
FYI, here is my documentation PR: https://github.com/rcsb/py-rcsbsearchapi/pull/47

File: rcsbsearchapi/__init__.py [hunk 1]
```diff
@@ -1,53 +1,21 @@
...
-from .search import Attr, Group, Query, Session, TextQuery, Value
+from typing import List
+from .search import Terminal, SCHEMA  # noqa: F401
+from .search import Attr, AttributeQuery, Group, Query, TextQuery
 
 __version__ = "1.6.0"
 
+# # loading rcsb_attributes can cause errors, so load it lazily
+# if TYPE_CHECKING:
+#     from .schema import SchemaGroup
```
[the remark this answers is not in the document]
Comment (ivana-truong) [reply rcsbsearchapi/__init__.py:11]: I am not 100% sure, but there aren't any circular import errors without these lines, so I think it's ok. As far as I could tell, that would be the reason to import with a TYPE_CHECKING check.

I'll go ahead and get rid of these lines!

File: rcsbsearchapi/search.py [hunk 2]
```diff
@@ -305,14 +328,304 @@ def _assign_ids(self, node_id=0) -> Tuple[Query, int]:
...
+    def __contains__(self, key):
+        return key in self.__dict__
+
+    def __getitem__(self, key):
+        if key in self.__dict__:
+            return self.__dict__[key]
+        raise KeyError(f"'{key}' not found in object")
+
+
+SCHEMA = Schema(Attr, "SchemaGroup")
```
[the remark this answers is not in the document]
Comment (ivana-truong) [reply rcsbsearchapi/search.py:619]: Yeah I tried as well, but ran into circular import errors and I don't think there's a way to solve this one. So, I'll leave it here.

**Pull request merged**